src/components/AddItemForm.js

```diff
@@ -16,23 +33,61 @@ const AddItemForm = () => {
     setNewItem(e.target.value);
   };
 
+  //Check for duplication:
+  //get the existing items list from firebase
+  //loop through the existing items list to check if there is a match with current item
+  //set itemExist to true if duplication and return it
+
+  const checkDuplication = (newItem) => {
+    let existingData = datas;
+
+    let itemExist = false;
+
+    existingData.forEach((object) => {
+      //Remove punctuation of existing item with regex
+      let existingItem = object.Item;
+      let cleanExistingItem = existingItem.replace(/[\W|_]/g, '');
+
+      //Remove punctuation of current item with regex
+      let currentItem = newItem;
+      let cleanCurrentItem = currentItem.replace(/[\W|_]/g, '');
+
+      //Check for duplication while normalizin capitalization
+      if (cleanCurrentItem.toLowerCase() === cleanExistingItem.toLowerCase()) {
+        itemExist = true;
+      }
+    });
+    return itemExist;
+  };
+
+  //Set error message and erase it after 3 sec and focus text input
+  const showErrorMessage = () => {
+    const newItemInput = document.querySelector('#newItem');
```

Comment:
You have used `querySelector`, I wonder if `useRef` could work here instead?

Reply:
I agree, `userRef` makes more sense here as we are using React and not vanilla JS.